node/src/main/scala/coop/rchain/node/NodeRuntime.scala

```diff
@@ -67,10 +67,12 @@ class NodeRuntime private[node] (
     eventLog: EventLog[Task]
 ) {
 
+  private[this] val numCPU = Math.max(java.lang.Runtime.getRuntime.availableProcessors(), 2)
+
   private[this] val loopScheduler =
     Scheduler.fixedPool("loop", 4, reporter = UncaughtExceptionLogger)
   private[this] val grpcScheduler =
-    Scheduler.cached("grpc-io", 4, 64, reporter = UncaughtExceptionLogger)
+    Scheduler.cached("grpc-io", 4, numCPU * 10, reporter = UncaughtExceptionLogger)
```

Comment:
This thread pool should be only for async IO operations with lower amount of threads. Is it used for something else like processing messages?
For reference, good description about different pool configurations.
https://gist.github.com/djspiewak/46b543800958cf61af6efa8e072bfd5c

Reply:
It is used for all GRPC related operation - kademlia p2p discovery, p2p messaging, GRPC transport both packets and streaming (which end up adding blocks). I guess there should be several schedulers for different things.

Reply:
@tgrospic does this make sense? https://github.com/rchain/rchain/pull/2902/commits/05ade87d830d969316784b9b2db4166f3435410a. As we had growing number of unconsumed block messages - I wonder if it is a good idea to dedicate scheduler for that particular task.

Reply:
Weird but I can see on one of 5 nodes growing number of unprocessed messages
```sh
pdsh -w ^sandhosts "ls -l /var/lib/rnode/tmp/comm/ | wc -l"
nzpr@node1: 1
nzpr@node0: 1
nzpr@node4: 1
nzpr@node2: 85
nzpr@node3: 1
```

Reply:
Let's keep `grpc-io` as it is. Before change, we must be sure that is only executing async IO. It was used to process incoming blocks so your change to another schedule will address this.

Reply:
We have enough schedulers, maybe even to many. They are all competing for fixed number of processors so there is no magic that can do. 
There is no reason to have more schedulers for async IO operations, or even more threads, because they are not taking CPU time. Instead, we can have opposite effect, many IO threads can interrupt main computation where context switching is much more expensive.

Reply:
Actually this commit made it worse, need to investigate.

Reply:
It made it worse because of this piece of code
https://github.com/rchain/rchain/pull/2902/commits/49f8f576da7754587e81cf38ea843d6ced08ecc8#diff-22a8d68f65fb447029ffa3431a23b529R366-R379
I pushed the fix https://github.com/rchain/rchain/pull/2902/commits/d02b9266f04e40bcf06f4f8e51795a4522cf47cd

Reply:
And this bad piece of code was less destructive when block messaging was handled on grpc-io scheduler. So I'm not sure anymore if we should push block handling to dedicated (or main) scheduler as per https://github.com/rchain/rchain/commit/05ade87d830d969316784b9b2db4166f3435410a

@tgrospic how do you think?

Reply:
> Either way, you're almost always going to have some sort of singleton object somewhere in your application which just has these three pools, pre-configured for use. If you ascribe to the "implicit ExecutionContext pattern", then you should make the CPU pool the implicit one, while the others must be explicitly selected.
From previous link but mentioned in the second also. :)

> If you are looking for less manual work - monix.eval.Task is great middleground which by default shifts tasks automatically in batches preserving both great throughput and decent latency off the shelf and exposes very rich configuration options if you have more advanced use case.
https://typelevel.org/cats-effect/concurrency/basics.html#thread-scheduling

Three schedulers, CPU bounded, implicit as we have main, async IO like our grpc-io and blocking, which we should not have because we use "shifting" for most flatMaps with monix Task.

Reply:
Ok, so I'm going to revert https://github.com/rchain/rchain/commit/05ade87d830d969316784b9b2db4166f3435410a